_user/rotrectangle/main.c

```diff
@@ -0,0 +1,273 @@
+	while ((self->stop == 0) && (ch = getchar())) {
+		if (ch == 27) {					/*ESC code from arrow keys*/
+			getchar();
+			ch = getchar();
+		}
+		switch (ch) {
+			case 67:
+				if (self->rec.xc < (self->maxxc-2))
+					self->rec.xc+=3;
+				break;
+			case 68:
+				if (self->rec.xc > (self->minxc+2))
+					self->rec.xc-=3;
+				break;
+			case 65:
+				if (self->rec.yc > (self->minyc+2))
+					self->rec.yc-=3;
+				break;
+			case 66:
+				if (self->rec.yc < (self->maxyc-2))
+					self->rec.yc+=3;
+				break;
+			case 43:
+				if (self->rotSpeed < 5)
+					self->rotSpeed++;
+				break;
+			case 45:
+				if (self->rotSpeed > 1)
+					self->rotSpeed--;
+				break;
+			case 'r':
+				self->rec.color = 4;
+				break;
+			case 'g':
+				self->rec.color = 2;
+				break;
+			case 'b':
+				self->rec.color = 1;
+				break;
+			case 'q':
+				self->stop = 1;
+				break;
+			default:
+				break;
+		}
+	}
+}
+
+static int rotrectangle_print(rectangle_t *self, graph_t *graphp, unsigned int angle)
+{
+	double dx, dy, angleRad, xcDouble, ycDouble, aDouble, bDouble;
+	unsigned int x0, y0, x, y;		/* (x,y) is start point for next rectangle lines */
+
+	xcDouble = (double)self->xc;
+	ycDouble = (double)self->yc;
+	aDouble = (double)self->a;
+	bDouble = (double)self->b;
+
+	angleRad = ((double)(angle)*3.141592)/180.0;
+	x0 = (unsigned int)(xcDouble - ( (sin(M_PI/2-angleRad-atan(bDouble/aDouble)))*sqrt(pow(aDouble, 2.0)+pow(bDouble, 2.0)) )/2.0);
+	y0 = (unsigned int)(ycDouble - ( (cos(M_PI/2-angleRad-atan(bDouble/aDouble)))*sqrt(pow(aDouble, 2.0)+pow(bDouble, 2.0)) )/2.0);
+
+	x = x0;
+	y = y0;
+	dx = (self->a)*cos(angleRad); 
+	dy = (self->a)*sin(angleRad);
+	graph_line(graphp, x, y, (int)dx, (int)dy, 1, self->color, GRAPH_QUEUE_HIGH);
+
+	x = x0+dx;
+	y = y0+dy;
+	dx = (self->b)*cos(angleRad+(0.5*3.141592));
+	dy = (self->b)*sin(angleRad+(0.5*3.141592));
+	graph_line(graphp, x, y, (int)dx, (int)dy, 1, self->color, GRAPH_QUEUE_HIGH);
+
+	x = x0;
+	y = y0;
+	dx = (self->b)*cos(angleRad+(0.5*3.141592)); 
+	dy = (self->b)*sin(angleRad+(0.5*3.141592));
+	graph_line(graphp, x, y, (int)dx, (int)dy, 1, self->color, GRAPH_QUEUE_HIGH);
+
+	x = x0+dx;
+	y = y0+dy;
+	dx = (self->a)*cos(angleRad); 
+	dy = (self->a)*sin(angleRad);
+	graph_line(graphp, x, y, (int)dx, (int)dy, 1, self->color, GRAPH_QUEUE_HIGH);
+
+	return 0;
+}
+
+static int rotrectangle_save(rotrectangle_t *self)
+{
+	self->toRemove.xc = self->rec.xc;
+	self->toRemove.yc = self->rec.yc;
+	self->toRemove.a = self->rec.a;
+	self->toRemove.b = self->rec.b;
+	self->toRemove.color = 0;
+
+	return 0;
+}
+
+static int rotrectangle_rotating(rotrectangle_t *self, graph_t *graphp)
+{
+	unsigned int alfa, ret;
+	ret = 0;
+	while (self->stop == 0) {
+		for (alfa=0; alfa<360; alfa+=self->rotSpeed) {
+			if (self->stop != 0)
+				break;
+			ret = rotrectangle_print(&(self->rec), graphp, alfa);
+			// graph_fill(graphp, self->rec.xc, self->rec.yc, self->rec.color, GRAPH_FILL_FLOOD, GRAPH_QUEUE_HIGH);
```

Comment:
Why is this line commented out?

Reply:
There is a problem with screen refreshing when I use filled rectangle (refresh occurs after displaying frame and before displaying filling) and Łukasz said I can comment it out now. So this is a demo without it.

Reply:
I think a comment about it would be helpful, so we won't forget why it is commented out for now

Reply:
Please comment this line using C comments  ( /* */ ) instead of C++ double /. Rotating empty rectangle is pretty enough for purposes of this demo.